### ISSUE TITLE
Username accepts upper case characters when creating an account

### ISSUE THREAD

**Issue by zhienzhien** [mattermost/mattermost-server#7921]
#### Summary
Username accepts upper case characters when creating an account in Mattermost.

#### Steps to reproduce
1. Open "http://demo.mattermost.com" in the browser. 
2.  Click 'Create one now' link to create an account
3. Enter data in the following fields:
- What's your email address?: **jhorellnerizon@gmail.com**
- Choose your username: **ZHIEN**
- Choose your password: ***********
4. Click 'Create Account' button

#### Expected behavior
- The account will not be created and an error message will be displayed stating "Username should contain only lowercase characters". It also must accept a combination of numbers, letters, and symbols.

#### Observed behavior (that appears unintentional) 
No error message was displayed by clicking 'Create Account' button. The account was successfully created and joined a channel. 
![image](https://user-images.githubusercontent.com/34127088/33429629-99b2a71c-d608-11e7-8fca-cfbe75222728.png)
![image](https://user-images.githubusercontent.com/34127088/33429633-9e914590-d608-11e7-9ec4-20dc19c0e32e.png)

#### Possible fixes
There should be an error message displayed prompting the user that the Username entered is invalid and must be corrected.

**Comment by lindy65** [comment]
Hi @zhienzhien,

Thanks for your feedback,

The username field is not case-sensitive.  This is by design.

Is there a specific reason you'd want the username to be created using only lower-case letters?

**Comment by zhienzhien** [comment]
Hi,

I just based my testing on the caption indicated under the Username field.
There's a caption indicated stating "Username must begin with a letter, and
contain between 3 to 22 *lowercase* characters made up of numbers, letters,
and the symbols '.', '-' and '_'.

It would be better if the lowercase word will be removed or indicate that
it's not case-sensitive.

Regards,
Jhorell

**Comment by lindy65** [comment]
Ah, you're quite correct @zhienzhien !

The help text is misleading.  I've created a [ticket](https://mattermost.atlassian.net/browse/PLT-8280) to fix the issue.

Just an FYI, even if you create a username with upper case letters, once you have logged in, the system converts your username to lower case letters in the UI

![image](https://user-images.githubusercontent.com/13020833/33540025-c87187f0-d8d1-11e7-8e8b-16dc15437710.png)

I created this account using lindy.UPPER and as you see in the image above, my username has been converted to lower case.

So, to summarize, account creation and log-in is not case sensitive - you can create an account with lower and/or uppercase letters and specify a username in upper case letters but, once logged in, your username is displayed in lower case.

**Comment by lindy65** [comment]
I'll close this issue off for now and we can track via the ticket linked above.

**Comment by DeanDeMarco** [comment]
> Hi, I just based my testing on the caption indicated under the Username field. There's a caption indicated stating "Username must begin with a letter, and contain between 3 to 22 *lowercase* characters made up of numbers, letters, and the symbols '.', '-' and '_'. It would be better if the lowercase word will be removed or indicate that it's not case-sensitive. Regards, Jhorell
> […](#)

Awesome.

**Comment by lindsayfowler** [comment]
Yes, but whilst it appears that username capitalization is not a problem, because it converts it back to lowercase...
It seemingly can cause hell when you use Github Pages to host a static site. At least I think so. 

The repo and the username need to be exactly the same, and so, at the very least, it causes confusion when naming the repo: 
do I name it **MyRepo.io** like my original username or **myrepo.io** like it was supposedly converted to? 

In my experience, naming it the latter in lowercase caused GitHub pages to behave unpredictably and it was very difficult to nut out what was going wrong. One image worked all others didn't. Then gradually all pages refused to load. I noticed upon creating the repo and  on the first push, everything worked, except images, and then as I made changes to the image links (people said I needed to use markdown - which was just untrue) the whole site fell apart and was unable to be revived.  One time in the bash prompt I noticed after a second upload a little message saying something to the effect that my files had moved to a location that was same name but different case... which got me thinking about the username I'd chosen a few years earlier. I had to delete my Github account and all my repos and create a brand new account (in lower case this time) then I uploaded the very same site that refused to work for months before, and with the username in lowercase everything was fine. Possibly a coincidence. I don't understand the processing that Github does on the files from its end - I mean it blabbers on about changing LF to CRLF but I'm sure it does more than just that. 

So perhaps something else going that I am unaware of, but I found that the original site which worked locally, with relative paths such as ../photos/abc.jpg worked fine once I set up a new account with lowercase letters.

**Comment by olafur-palsson** [comment]
I stumbled upon this issue, I believe that having upper-case letters allowed can have compatibility issues down the line for GitHub. Colleague of mine wants to use his name `npm publish` and `npm login` when using the GitHub packages feature. It's not possible as of now.

**Comment by JohnnyGagnier** [comment]
Our yearly Game Off is a month-long game jam where participants use their preferred game engines, libraries, and programming languages to develop games. All levels of game developers are welcome to join[,](https://mybankinginformation.com/does-winco-accept-ebt/) from amateurs to professionals. It's a wonderful justification for picking up a new skill, working on a project over the weekend with friends, or making your first game all by yourself!

**Comment by JohnnyGagnier** [comment]
Make a Wish [url=https://pixellogo.com/blogs/pixellogo-blog-1/new-make-a-wish-logo]Shirt(/url) is a fun way to support kids in making their dreams come true!
